### PR TITLE
feat(actions): add addedWithVersion to CommonSettings for version-dependent defaults (#315)

### LINKED ISSUE #315: feat(actions): add addedWithVersion to CommonSettings for version-dependent defaults
## What would you like?

Add an `addedWithVersion` field to `CommonSettings` that records the plugin version when an action instance is first placed on the Stream Deck. This enables version-dependent setting defaults — new instances get modern defaults while existing instances preserve backward-compatible behavior.

### How it works

1. `CommonSettings` gets an `addedWithVersion` field, defaulting to `"0.0.0"` when missing (so existing instances are always treated as "older than everything")
2. On first `onWillAppear`, if `addedWithVersion` is absent, the current plugin version (from the build-time config file, #314) is persisted via `setSettings()`
3. Settings with version-dependent defaults compare `addedWithVersion` against a hardcoded threshold:

```typescript
const defaultMode = semver.lt(settings.addedWithVersion, "1.13.0") ? "toggle" : "select";
```

### Dependencies

- Depends on #314 (build-time config file) for reading the current plugin version at runtime

## Use case

As the plugin evolves, new settings are added that should have different defaults for new users vs. existing users. Without version tracking, there's no way to distinguish existing action instances (which should preserve old behavior) from newly added ones (which should get improved defaults).

First consumer: tire service toggle mode (#313).

## Area

Action (existing)

## Affected files

- `packages/deck-core/src/common-settings.ts` — add `addedWithVersion` field
- `packages/deck-core/src/base-action.ts` or `connection-state-aware-action.ts` — persist version on first appear
- `packages/actions/` — any action tests that mock CommonSettings

## Acceptance criteria

- [ ] `addedWithVersion` field added to `CommonSettings` with `"0.0.0"` default
- [ ] Version is persisted on first `onWillAppear` when missing
- [ ] Version is read from the build-time config file (#314)
- [ ] Existing action instances (no `addedWithVersion` stored) get `"0.0.0"`
- [ ] New action instances get the current plugin version
- [ ] Update tests to cover version persistence

### PR DESCRIPTION
**Pull request by niklam**
## Related Issue

Fixes #315

## What changed?

- Added `addedWithVersion` field to `CommonSettings` (Zod default `"0.0.0"` for pre-existing instances)
- `BaseAction.onWillAppear()` persists the current plugin version (from `config.json` via `getPluginVersion()`) on first appearance when the field is missing
- Existing instances get `"0.0.0"`, new instances get the current plugin version (e.g., `"1.13.0"`)
- This enables version-dependent setting defaults: `semver.lt(settings.addedWithVersion, "1.13.0") ? oldDefault : newDefault`

## How to test

1. Run `pnpm test` — all tests pass including 3 new tests for version persistence
2. Verify behavior: action with no `addedWithVersion` in settings → persisted on first `onWillAppear`
3. Verify behavior: action with existing `addedWithVersion` → not overwritten
4. Verify behavior: plugin config not initialized → no persistence attempted

## Checklist

- [x] Linked to an approved issue
- [x] New code has unit tests
- [x] Changes registered in all applicable plugins (Stream Deck, Mirabox)
- [x] No unrelated changes included

## Summary by CodeRabbit

* **New Features**
  * Actions automatically capture and persist the plugin version when first added to a device, improving version tracking and visibility across different plugin deployments.

* **Tests**
  * Added test coverage for the automatic version tracking functionality, including verification of initialization behavior and edge cases.

### CODE REVIEW

File: packages/deck-core/src/base-action.ts [hunk 1]
```diff
@@ -290,13 +291,22 @@ export abstract class BaseAction<T = Record<string, unknown>> implements IDeckAc
...
     const settings = ev.payload.settings as Record<string, unknown>;
 
+    // Persist addedWithVersion on first appearance (missing = pre-existing instance).
+    // Note: setSettings triggers onDidReceiveSettings — benign since it only re-reads
+    // the same settings that were just written.
+    if (!settings.addedWithVersion && isPluginConfigInitialized()) {
+      const version = getPluginVersion();
+      this.logger.debug(`Persisting addedWithVersion=${version} for context ${ev.action.id}`);
+      await ev.action.setSettings({ ...settings, addedWithVersion: version });
+    }
```
[the remark this answers is not in the document]
Comment (niklam) [reply packages/deck-core/src/base-action.ts:312]: Fixed in 0e65bc2f — wrapped in try/catch so flag overlay setup is not skipped on failure.
